apps/pie-docs/src/assets/js/scroll-to-top.js

```diff
@@ -0,0 +1,45 @@
+let scrollToTopButton;
+
+const initialiseScrollToTop = () => {
+    scrollToTopButton = document.querySelector('[data-js="scroll-to-top"]');
+    const siteHeader = document.querySelector('[data-js="site-header"]');
+
+    if (!scrollToTopButton) return;
+
+    const scrollObserver = new IntersectionObserver((entries) => {
+        entries.forEach((entry) => {
+            calculateButtonVisibility(entry.isIntersecting);
+        });
+    });
+
+    if (siteHeader) {
+        scrollObserver.observe(siteHeader);
+    }
```

Comment:
do you think we should unobserve / disconnect when the page unloads?

Reply:
Interesting question, I think the browser's GC should handle it by default. I think it only becomes an issue when you have something like an SPA?